src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysisContext.cs

```diff
@@ -0,0 +1,138 @@
+            bool isNullPropertyFlagged,
+            ImmutableHashSet<string> methodNamesToCheckForFlaggedUsage)
+            : base(valueDomain, wellKnownTypeProvider, controlFlowGraph, owningSymbol, interproceduralAnalysisKind, pessimisticAnalysis,
+                  predicateAnalysis: false, copyAnalysisResultOpt: null, pointsToAnalysisResultOpt: pointsToAnalysisResultOpt,
+                  getOrComputeAnalysisResult: getOrComputeAnalysisResult,
+                  parentControlFlowGraphOpt: parentControlFlowGraphOpt,
+                  interproceduralAnalysisDataOpt: interproceduralAnalysisDataOpt)
+        {
+            this.TypeToTrackMetadataName = typeToTrackMetadataName;
+            this.IsNewInstanceFlagged = isNewInstanceFlagged;
+            this.PropertyToSetFlag = propertyToSetFlag;
+            this.IsNullPropertyFlagged = isNullPropertyFlagged;
+            this.MethodNamesToCheckForFlaggedUsage = methodNamesToCheckForFlaggedUsage;
+        }
+
+        public static PropertySetAnalysisContext Create(
+            AbstractValueDomain<PropertySetAbstractValue> valueDomain,
+            WellKnownTypeProvider wellKnownTypeProvider,
+            ControlFlowGraph controlFlowGraph,
+            ISymbol owningSymbol,
+            InterproceduralAnalysisKind interproceduralAnalysisKind,
+            bool pessimisticAnalysis,
+            PointsToAnalysisResult pointsToAnalysisResultOpt,
+            Func<PropertySetAnalysisContext, PropertySetAnalysisResult> getOrComputeAnalysisResult,
+            string typeToTrackMetadataName,
+            bool isNewInstanceFlagged,
+            string propertyToSetFlag,
+            bool isNullPropertyFlagged,
+            ImmutableHashSet<string> methodNamesToCheckForFlaggedUsage)
+
+        {
+            return new PropertySetAnalysisContext(
+                valueDomain, wellKnownTypeProvider, controlFlowGraph, owningSymbol, interproceduralAnalysisKind,
+                pessimisticAnalysis, pointsToAnalysisResultOpt, getOrComputeAnalysisResult, null,
+                null,
+                typeToTrackMetadataName,
+                isNewInstanceFlagged,
+                propertyToSetFlag,
+                isNullPropertyFlagged,
+                methodNamesToCheckForFlaggedUsage);
+        }
+
+        public override PropertySetAnalysisContext ForkForInterproceduralAnalysis(
+            IMethodSymbol invokedMethod,
+            ControlFlowGraph invokedCfg,
+            IOperation operation,
+            PointsToAnalysisResult pointsToAnalysisResultOpt,
+            CopyAnalysisResult copyAnalysisResultOpt,
+            InterproceduralBinaryFormatterAnalysisData interproceduralAnalysisData)
+        {
+            Debug.Assert(pointsToAnalysisResultOpt != null);
+            Debug.Assert(copyAnalysisResultOpt == null);
+
+            return new PropertySetAnalysisContext(
+                ValueDomain, WellKnownTypeProvider, invokedCfg, invokedMethod, InterproceduralAnalysisKind.ContextSensitive,
```

Comment:
>InterproceduralAnalysisKind.ContextSensitive [](start = 79, length = 44)

We should not for context sensitive analysis if the client request none or non-context sensitive. You should either use the [kind from the existing context](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Utilities/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysisContext.cs#L61) or can force None if you want interprocedural analysis only [one level down](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Utilities/FlowAnalysis/Analysis/ParameterValidationAnalysis/ParameterValidationAnalysisContext.cs#L70)